src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -722,10 +722,14 @@ int32_t NaClSysOpen(struct NaClAppThread  *natp,
   retval = NaClHostDescOpen(hd, path, flags, mode);
   NaClLog(1, "Cage %d NaClHostDescOpen(0x%08"NACL_PRIxPTR", %s, 0%o, 0%o) returned %d\n",
           nap->cage_id, (uintptr_t) hd, path, flags, mode, retval);
-  if (!retval) {
+  if (retval >= 0) {
     retval = NaClSetAvail(nap, ((struct NaClDesc *) NaClDescIoDescMake(hd)));
     NaClLog(1, "Entered into open file table at %d\n", retval);
   }
+  else {
+    NaClLog(1, "Open returned error %d\n", retval);
+    return -NACL_ABI_EPERM;
+  }
```

Comment:
I'd put this above where you currently have your if block and remove the if statement, so something like
```
if(retval < 0) {
    NaClLog(1, "Open returned error %d\n", retval);
    return -NACL_ABI_EPERM;
}
retval = NaClSetAvail(nap, ((struct NaClDesc *) NaClDescIoDescMake(hd)));
NaClLog(1, "Entered into open file table at %d\n", retval);
```
Not a big deal but seems a little cleaner to me.